versioned_hdf5/backend.py

```diff
@@ -5,6 +5,7 @@
 from .hashtable import Hashtable
 
 DEFAULT_CHUNK_SIZE = 2**12
+DATA_VERSION = 'v2'
```

Comment:
Let's use an integer instead of a string, to make comparisons easier.

Reply:
I like it. I considered moving to `packaging.version.Version`, but I think this is just simpler.

Reply:
File versions are usually just simple integers. There's no need to be semantic here.